article/repository/mysql_article.go

```diff
@@ -82,44 +81,43 @@ func (m *mysqlArticleRepository) Fetch(ctx context.Context, cursor string, num i
 	return res, nextCursor, err
 
 }
-func (m *mysqlArticleRepository) GetByID(ctx context.Context, id int64) (*models.Article, error) {
+func (m *mysqlArticleRepository) GetByID(ctx context.Context, id int64) (res domain.Article, err error) {
 	query := `SELECT id,title,content, author_id, updated_at, created_at
   						FROM article WHERE ID = ?`
 
 	list, err := m.fetch(ctx, query, id)
 	if err != nil {
-		return nil, err
+		return domain.Article{}, err
 	}
 
-	a := &models.Article{}
+	res = domain.Article{}
 	if len(list) > 0 {
-		a = list[0]
+		res = list[0]
 	} else {
-		return nil, models.ErrNotFound
+		return res, domain.ErrNotFound
 	}
-
-	return a, nil
+	return
 }
 
-func (m *mysqlArticleRepository) GetByTitle(ctx context.Context, title string) (*models.Article, error) {
+func (m *mysqlArticleRepository) GetByTitle(ctx context.Context, title string) (res domain.Article, err error) {
 	query := `SELECT id,title,content, author_id, updated_at, created_at
   						FROM article WHERE title = ?`
 
 	list, err := m.fetch(ctx, query, title)
 	if err != nil {
-		return nil, err
+		return res, err
 	}
 
-	a := &models.Article{}
+	res = domain.Article{}
 	if len(list) > 0 {
-		a = list[0]
+		res = list[0]
 	} else {
-		return nil, models.ErrNotFound
+		return res, domain.ErrNotFound
 	}
-	return a, nil
+	return
```

Comment:
You're not using pointer and you change method to return nothing. is it correct?

Reply:
@jojoarianto actually @bxcodec using named return, so at the end of function, you need to put return
and for pointer afaik is not good using to many pointer, because you will using heap memory instead of stack and will allocated more